theano/gpuarray/basic_ops.py

```diff
@@ -1658,7 +1658,12 @@ def c_code(self, node, name, inp, out, sub):
 
         ls = 1;
         gs = 256;
-        err = eye_call(1, &gs, &ls, 0, %(z)s->ga.data, dims[0], dims[1], k);
+        col_off = (size_t) (k > 0?k:0);
+        row_off = (size_t) (k < 0?-k:0);
+        if (row_off < dims[0] && col_off < dims[1]) {
+            err = eye_call(1, &gs, &ls, 0, %(z)s->ga.data, dims[0], dims[1], k);
+        }
+
         if (err != GA_NO_ERROR) {
```

Comment:
No error there, because err has already been initialised, probably to the same value as `GA_NO_ERROR`. But you're right it doesn't make sense, I should move it in.